File: packages/java/src/main/java/ai/alumnium/tool/SwitchToNextTabTool.java
package ai.alumnium.tool;

import ai.alumnium.driver.BaseDriver;
import ai.alumnium.tool.annotation.ToolDescription;

@ToolDescription("Switch to the next browser tab/window.")
public record SwitchToNextTabTool() implements BaseTool {
  @Override
  public void invoke(BaseDriver driver) {
    driver.switchToNextTab();
  }
}

File: packages/typescript/src/tools/SwitchToNextTabTool.ts
import { BaseDriver } from "../drivers/BaseDriver.ts";
import { BaseTool } from "./BaseTool.ts";

export class SwitchToNextTabTool extends BaseTool {
  static description = `Switch to the next browser tab/window.`;

  async invoke(driver: BaseDriver): Promise<void> {
    await driver.switchToNextTab();
  }
}
